File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict

def solution(clothes):
    
    hash=defaultdict(int)
    
    for item,name in clothes:
        if name in hash:
            hash[name]+=1
        else:
            hash[name]=1
    
    answer = 1
    for cnt in hash.values():
        answer*=(cnt+1)
    
    return answer-1

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>
using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    
    unordered_map<string,int> wearing;
    for(auto x:clothes){
        wearing[x[1]]++;
    }
    
    for(auto x:wearing){
        x.second++;
        answer=answer*x.second;
    }
    
    
    return answer-1;
}